src/components/Dashboard/Role/AddUser.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { Button, Form } from 'react-bootstrap';
+import {
+  DashboardLink as DashLink,
+  DashboardLayout as Layout
+} from 'components/Dashboard/';
+import { UserRoles } from 'components/Dashboard/';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { actions as rolesActions } from 'reducers/roles';
+import { actions as dashActions } from 'reducers/dashboard';
+import { actions as usersActions } from 'reducers/users';
+import { getAllUsers } from 'selectors/users';
+
+export class RoleAddUser extends Component {
+  static propTypes = {
+    actions: PropTypes.object.isRequired,
+    name: PropTypes.string,
+    layoutOptions: PropTypes.object.isRequired,
+    roleId: PropTypes.number,
+    users: PropTypes.array
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = { userId: '0', submitting: false };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const { actions } = this.props;
+
+    actions.requestUsers();
+  }
+
+  handleChange(e) {
+    const { actions } = this.props;
+
+    this.setState({ userId: e.target.value });
+    // Refresh User Roles for <UserRoles /> component
+    actions.requestUserRoles({ userId: e.target.value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.setState({ submitting: true });
+    const { userId } = this.state;
+    const { actions, roleId } = this.props;
+
+    if (userId === '0') {
+      this.setState({ submitting: false });
+    } else {
+      actions.createRoleUser({ roleId, userId, active: true });
+      actions.selectDashboard({ name: 'Roles' });
+    }
+  }
+
+  render() {
+    const { users, name, layoutOptions, roleId } = this.props;
+    const { userId, submitting } = this.state;
+
+    return (
+      <Layout
+        pageTitle="Add Role - Dashboard"
+        header={`Roles > Assign ${name} Role`}
+        options={layoutOptions}
+      >
+        <FontAwesomeIcon icon="chevron-left" /> &nbsp;
+        <DashLink to="#roles" name="Roles">
+          Back
+        </DashLink>
+        <Form noValidate onSubmit={this.handleSubmit}>
+          <Form.Group>
+            <Form.Label>Select User to Assign to {name}s Role</Form.Label>
+            <Form.Control
+              as="select"
+              name="userId"
+              onChange={e => this.handleChange(e)}
+              value={userId}
+            >
+              <option value="0" key="x">
+                Choose a User
+              </option>
+              {users.map((user, index) => {
+                return (
+                  <option value={user.id} key={index}>
+                    {user.emailAddress}
+                  </option>
+                );
+              })}
+            </Form.Control>
+          </Form.Group>
+          {userId !== '0' && (
```

Comment:
Usually when I need to make one assertion in multiple JSX expressions I throw it up at the top of the render method as a const, makes things easier to refactor:

```jsx
const isUserIdSelected = userId !== '0';
```

And if the check is more complicated than just one or two clauses I'll throw it into a getter like:

```jsx
get userIdSelected() {
  const { something, another, third } = this.props;
  return (something && another) || (third && !something);
}

...

{this.userIdSelected ? <Something /> : <TheOtherThing />}
```

Reply:
Updated